src/org/mirah/typer/typer.mirah

```diff
@@ -574,6 +569,7 @@ class Typer < SimpleNodeVisitor
     end
     enclosing_node = node.findAncestor {|n| n.kind_of?(MethodDefinition) || n.kind_of?(Script)}
     if enclosing_node.kind_of? MethodDefinition
+      return nil if isMethodInBlock(MethodDefinition(enclosing_node)) # return types are not supported currently for methods which act as templates
```

Comment:
They do not need the future at all, because the MethodDefinition subtrees are currently just use as a template, and clone()d into new MethodDefinitions, which are then properly infer()ed on.

I've tried to replace the cloning by mere relinking of the AST, but failed in the first attempt due to compiler intricacies. As anonymous classes are currently a not-often-used feature, I think that this optimization can be implemented later.